channel/consensus_channel/consensus_channel.go

```diff
@@ -359,7 +378,12 @@ func (p *Proposal) equal(q *Proposal) bool {
 
 // isAddProposal returns true if the proposal contains a non-nil toAdd and a nil toRemove.
 func (p *Proposal) isAddProposal() bool {
-	return p.ToAdd != Add{} && p.ToRemove == struct{}{}
+	return p.ToAdd != Add{} && p.ToRemove == Remove{}
+}
+
+// isRemoveProposal returns true if the proposal contains a non-nil toRemove and a nil toAdd.
+func (p *Proposal) isRemoveProposal() bool {
+	return p.ToAdd == Add{} && p.ToRemove != Remove{}
 }
```

Comment:
Is it better for `Proposal` to implement `Type()`?

This way, someone can switch on `p.Type()` or check `p.Type() == AddType`